packages/node/src/transports/base/index.ts

```diff
@@ -181,10 +198,111 @@ export abstract class BaseTransport implements Transport {
-  protected async _send(
-    sentryRequest: SentryRequest,
-    originalPayload?: Event | Session | SessionAggregates,
-  ): Promise<Response> {
+  protected _createTask(sentryRequest: SentryRequest): () => Promise<Response> {
+    return () =>
+      new Promise<Response>((resolve, reject) => {
+        if (!this.module) {
+          throw new SentryError('No module available');
+        }
+        const options = this._getRequestOptions(this.urlParser(sentryRequest.url));
+        const req = this.module.request(options, res => {
+          const statusCode = res.statusCode || 500;
+          const status = Status.fromHttpCode(statusCode);
+
+          res.setEncoding('utf8');
+
+          /**
+           * "Key-value pairs of header names and values. Header names are lower-cased."
+           * https://nodejs.org/api/http.html#http_message_headers
+           */
+          let retryAfterHeader = res.headers ? res.headers['retry-after'] : '';
+          retryAfterHeader = (Array.isArray(retryAfterHeader) ? retryAfterHeader[0] : retryAfterHeader) as string;
+
+          let rlHeader = res.headers ? res.headers['x-sentry-rate-limits'] : '';
+          rlHeader = (Array.isArray(rlHeader) ? rlHeader[0] : rlHeader) as string;
+
+          const headers = {
+            'x-sentry-rate-limits': rlHeader,
+            'retry-after': retryAfterHeader,
+          };
+
+          const limited = this._handleRateLimit(headers);
+          if (limited)
+            logger.warn(
+              `Too many ${sentryRequest.type} requests, backing off until: ${this._disabledUntil(sentryRequest.type)}`,
+            );
+
+          if (status === Status.Success) {
+            resolve({ status });
+          } else {
+            let rejectionMessage = `HTTP Error (${statusCode})`;
+            if (res.headers && res.headers['x-sentry-error']) {
+              rejectionMessage += `: ${res.headers['x-sentry-error']}`;
+            }
+            reject(new SentryError(rejectionMessage));
+          }
+
+          // Force the socket to drain
+          res.on('data', () => {
+            // Drain
+          });
+          res.on('end', () => {
+            // Drain
+          });
+        });
+        req.on('error', reject);
+        req.end(sentryRequest.body);
+      });
+  }
+
+  /** JSDoc */
+  protected _queueRequest(sentryRequest: SentryRequest, originalPayload?: OriginalRequestPayload): void {
+    let requestType = sentryRequest.type;
+    // @ts-ignore just why...
+    if (requestType === 'sessions') {
+      requestType = 'session';
+    }
+
+    const queue = this._queues[requestType];
+
+    if (!queue) {
+      throw new SentryError(`Incorrect queue type ${requestType}`);
+    }
+
+    try {
+      queue.enqueue([sentryRequest, originalPayload]);
+    } catch (e) {
+      throw new SentryError(`Error enqueueing ${requestType} request: ${(e as Error).message}`);
+    }
+  }
+
+  /** JSDoc */
+  protected _yieldQueue(): Queue<RequestsQueue> {
```

Comment:
Thinking about using an iterator to move the responsibility from the queue management away and only focus on the transport. What do you think?